### PR TITLE
Refactor vertex buffer stuff

### PR DESCRIPTION
**Pull request by enbrain**
@apple502j This PR overwrites some of your recent names. I'd like to hear your thoughts.

### CODE REVIEW

**Review by apple502j — APPROVED** [review]
Good rename I think

**Comment by apple502j** [comment]
It would be nice if there was a javadoc describing the basics of OpenGL so rendering noobs like me can map stuff properly, like uploading 😀

File: mappings/com/mojang/blaze3d/systems/RenderSystem.mapping [hunk 1]
```diff
@@ -253,30 +253,36 @@ CLASS com/mojang/blaze3d/systems/RenderSystem
 		ARG 2 width
 		ARG 3 height
 	CLASS class_5590 IndexBuffer
```
Comment (Ueaj-Kerman) [inline mappings/com/mojang/blaze3d/systems/RenderSystem.mapping]:
```suggestion
	CLASS class_5590 ShapeElementBuffer
```
Maybe that's too verbose but I think IndexBuffer doesn't properly communicate that this holds/generates a pre-made element buffer for a specific shape. `PolygonElementBuffer` works too

File: mappings/com/mojang/blaze3d/systems/RenderSystem.mapping [hunk 2]
```diff
@@ -253,30 +253,36 @@ CLASS com/mojang/blaze3d/systems/RenderSystem
...
-		METHOD method_43409 isSizeLessThanOrEqual (I)Z
-			ARG 1 size
+		METHOD method_43409 isLargeEnough (I)Z
+			ARG 1 requiredSize
 		METHOD method_43410 bindAndGrow (I)V
-			ARG 1 newSize
+			COMMENT Binds this buffer as a current index buffer. If necessary, it grows this
+			COMMENT buffer in size and uploads indices to the corresponding buffer in GPU.
+			ARG 1 requiredSize
 		CLASS class_5591 IndexMapper
```
Comment (Ueaj-Kerman) [inline mappings/com/mojang/blaze3d/systems/RenderSystem.mapping]:
```suggestion
		CLASS class_5591 Triangulator
```
The process of converting a shape into triangles is called [triangulation](https://en.wikipedia.org/wiki/Polygon_triangulation) (confusingly, but it is the correct terminology)

**Review by Ueaj-Kerman — APPROVED** [review]

File: mappings/com/mojang/blaze3d/systems/RenderSystem.mapping (hunk 1 above)
Comment (enbrain) [reply mappings/com/mojang/blaze3d/systems/RenderSystem.mapping]:
```suggestion
	CLASS class_5590 ShapeIndexBuffer
```
While it's an element buffer in OpenGL term, "element" is already used for `VertexFormatElement` (vertex attribute) in yarn. I think "index buffer" is fine for now.

Comment (Ueaj-Kerman) [reply mappings/com/mojang/blaze3d/systems/RenderSystem.mapping]: That works too

**Comment by apple502j** [comment]
@enbrain Seems like this conflicted with the buffer builder PR?

**Review by Player3324 — APPROVED** [review]

**Comment by enbrain** [comment]
I will redo this PR.

**Pull request closed without merging**